event-processing/examples/service-integration-example-template.yml

```diff
@@ -0,0 +1,166 @@
+
+Resources:
+  AuditEventQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !Ref AuditEventQueueEncryptionKeyAlias
+      RedriveAllowPolicy:
+        redrivePermission: denyAll
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt AuditEventDeadLetterQueue.Arn
+        maxReceiveCount: 10
+      Tags:
+        - Key: Name
+          Value: !Join
+            - "-"
+            - - !Ref AWS::StackName
+              - "auditEventQueue"
+        - Key: Service
+          Value: "ci/cd"
+        - Key: Source
+          Value: "alphagov/path/to/your/template.yaml"
+
+  AuditEventQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref AuditEventQueue
+      PolicyDocument:
+        Statement:
+          - Sid: "AllowReadByTXMAAccount"
+            Effect: Allow
+            Principal:
+              AWS: !FindInMap [TxMAAccountARN, Environment, !Ref 'Environment']
+            Action:
+              - "sqs:ReceiveMessage"
+              - "sqs:DeleteMessage"
+              - "sqs:GetQueueAttributes"
+            Resource: !GetAtt AuditEventQueue.Arn
+
+  AuditEventDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !Ref AuditEventQueueEncryptionKeyAlias
+      Tags:
+        - Key: Name
+          Value: !Join
+            - "-"
+            - - !Ref AWS::StackName
+              - "auditEventDeadLetterQueue"
+        - Key: Service
+          Value: "ci/cd"
+        - Key: Source
+          Value: "alphagov/path/to/your/template.yaml"
+
+  AuditEventQueueEncryptionKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Symmetric key used to encrypt audit messages at rest in SQS
+      EnableKeyRotation: true
+      KeySpec: SYMMETRIC_DEFAULT
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: 'Enable Root access'
+            Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - 'kms:*'
+            Resource: '*'
+          - Sid: 'Allow decryption of events by TXMA'
+            Effect: Allow
+            Principal:
+              AWS: !FindInMap [TxMAAccountARN, Environment, !Ref 'Environment']
+            Action:
+              - 'kms:decrypt'
+            Resource: '*'
+      Tags:
+        - Key: Name
+          Value: !Join
+            - "-"
+            - - !Ref AWS::StackName
+              - "auditEventQueueEncryptionKey"
+        - Key: Service
+          Value: "ci/cd"
+        - Key: Source
+          Value: "alphagov/path/to/your/template.yaml"
+
+  AuditEventQueueEncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub alias/${AWS::StackName}/auditEventQueueEncryptionKey
+      TargetKeyId: !Ref AuditEventQueueEncryptionKey
+
+  AuditEventQueueConsumerRole:
+    Type: AWS::IAM::Role
+    Properties:
+      PermissionsBoundary: !Ref PermissionsBoundary

```

Comment:
Should we add a generic UsePermissionsBoundary condition?

Reply:
Because this is what is used in all the templates I've seen (I think it's so that, if there is no boundary, it can still deploy)